OpenRobertaRobot/src/main/java/de/fhg/iais/roberta/syntax/MotionParam.java

```diff
@@ -47,10 +50,33 @@ public String toString() {
          * @return reference returned so calls can be chained
          */
         public Builder<V> speed(Expr<V> speed) {
-            this.speed = speed;
+//            this.speed = speed;
+            this.speed = adjustSpeed(speed);
             return this;
         }
 
+        public Expr<V> adjustSpeed(Expr<V> speed){
+            try{
+                BigDecimal val = new BigDecimal(((NumConst)speed).getValue());
+                String outVal;
+
+                if(val.compareTo(new BigDecimal(100)) == 1){
+                    outVal = "100";
+                }
+                else if (val.compareTo(new BigDecimal(-100)) == -1){
+                    outVal = "100";
+                }
+                else{
+                    outVal = val.toString();
+                }
+
+                return NumConst.make(outVal, speed.getProperty(), speed.getComment());
+            }
+            catch(Exception e){
```

Comment:
Do not use a catch all exception, only catch the exception that may occur here.

Reply:
When I was testing it, I found that multiple types of expressions were being acautght. I originally only caught null pointer but It still produced errors.